lib/reporting/queries/joins.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Reporting
+  module Queries
+    module Joins
+      def joins_order
+        reflect query.join(association(Spree::LineItem, :order))
```

Comment:
Yeah it's a nice method!
```ruby
def reflect(query)
  dup.tap { |obj| obj.instance_variable_set('@query'.freeze, query) }
end
```
A simplified version of this would just be: 
```ruby
def reflect(query)
  self.class.new(query)
end
```
but the use of `#dup` allows any other instance variables that might be on the object to be maintained when the object is being recreated/modified.

Reply:
Some notes on the concept of Reflection, if anyone's interested: https://en.wikipedia.org/wiki/Reflective_programming